ci/run-sanity.sh

```diff
@@ -5,9 +5,26 @@ cd "$(dirname "$0")/.."
 # shellcheck source=multinode-demo/common.sh
 source multinode-demo/common.sh
 
+if [[ -z $CI ]]; then
+  # Build eargerly if needed for local development. Otherwise, odd timing error occurs...
+  $solana_keygen --version
+  $solana_genesis --version
+  $solana_faucet --version
+  $solana_cli --version
+  $agave_validator --version
+  $solana_ledger_tool --version
+fi
+
 rm -rf config/run/init-completed config/ledger config/snapshot-ledger
 
-SOLANA_RUN_SH_VALIDATOR_ARGS="--full-snapshot-interval-slots 200" timeout 120 ./scripts/run.sh &
+# Sanity-check that agave-validator can successfully terminate itself without relying on
+# process::exit() by extending the timeout...
+# Also the banking_tracer thread needs some extra time to flush due to
+# unsynchronized and buffered IO.
+SOLANA_RUN_SH_VALIDATOR_ARGS="${SOLANA_RUN_SH_VALIDATOR_ARGS} --full-snapshot-interval-slots 200" \
+  SOLANA_VALIDATOR_EXIT_TIMEOUT="${SOLANA_VALIDATOR_EXIT_TIMEOUT:-3600}" \
+  timeout 120 ./scripts/run.sh &
```

Comment:
yeah, this is overkill indeed after some thought: f9535ae64c1

because we're not passing `--preserve-status` to `timeout`, ci always will be flagged as red just with 120 even if stuck validator process ignores TERM signal and exit(0) after SOLANA_VALIDATOR_EXIT_TIMEOUT.